File: VB/T148944/Service/PrintDialogService.vb
Imports System.Printing
Imports System.Windows.Controls
Imports DevExpress.Mvvm.UI

Namespace T148944.Service
    Public Class PrintDialogService
        Inherits ServiceBase
        Implements IPrintDialogService

        Private ReadOnly dialog As New PrintDialog()

        Public ReadOnly Property PrintQueue() As PrintQueue Implements IPrintDialogService.PrintQueue
            Get
                Return dialog.PrintQueue
            End Get
        End Property

        Public Function ShowDialog() As Boolean Implements IPrintDialogService.ShowDialog
            Return dialog.ShowDialog() = True
        End Function
    End Class
End Namespace

File: CS/T148944/Service/PrintDialogService.cs
using System.Printing;
using System.Windows.Controls;
using DevExpress.Mvvm.UI;

namespace T148944.Service {
    public class PrintDialogService : ServiceBase, IPrintDialogService {
        readonly PrintDialog dialog = new PrintDialog();

        public PrintQueue PrintQueue {
            get { return dialog.PrintQueue; }
        }

        public bool ShowDialog() {
            return dialog.ShowDialog() == true;
        }
    }
}
